---
title: "Lung Metagenomic Analysis"
author: "Virly Ananda"
output: 
  github_document:
    fig_width: 5
    fig_height: 5
    dev: jpeg
---
Installing packages  
```{r}
# installing bioconductor package "metagenomeSeq"
#if (!requireNamespace("BiocManager", quietly = TRUE))
  #  install.packages("BiocManager")

#BiocManager::install("metagenomeSeq")

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("biomformat")
```

Loading libraries  
```{r}
library(metagenomeSeq)
library(biomformat)
```

Biom-Format  
```{r}
biom_file <- system.file("extdata", "min_sparse_otu_table.biom",
package = "biomformat")
b <- read_biom(biom_file)
biom2MRexperiment(b)
```

As an example, we show how one can write a MRexperiment object out as a BIOM file.
Here is an example writing out the mouseData MRexperiment object to a BIOM file.
```{r}
# data(mouseData)
# # options include to normalize or not
# b <- MRexperiment2biom(mouseData)
# write_biom(b, biom_file = "Ëœ/Desktop/otu_table.biom")
```

Loading count data
```{r}
dataDirectory <- system.file("extdata", package = "metagenomeSeq")
lung = loadMeta(file.path(dataDirectory, "CHK_NAME.otus.count.csv"))
dim(lung$counts)
```

Loading taxonomy
```{r}
taxa = read.delim(file.path(dataDirectory, "CHK_otus.taxonomy.csv"), stringsAsFactors = FALSE)
```

Loading metadata: 
Phenotype data can be optionally loaded into R with loadPhenoData. This function loads the data as a list.  
```{r}
clin = loadPhenoData(file.path(dataDirectory, "CHK_clinical.csv"),
tran = TRUE)
ord = match(colnames(lung$counts), rownames(clin))
clin = clin[ord, ]
head(clin[1:2, ])
```

Creating MRexperiment object  
```{r}
phenotypeData = AnnotatedDataFrame(clin)
phenotypeData
```

A feature annotated data frame. In this example it is simply the OTU numbers, but it can as easily be the annotated taxonomy at multiple levels.

```{r}
OTUdata = AnnotatedDataFrame(taxa)
OTUdata
```

```{r}
obj = newMRexperiment(lung$counts,phenoData=phenotypeData,featureData=OTUdata)
# Links to a paper providing further details can be included optionally.
# experimentData(obj) = annotate::pmid2MIAME("21680950")
obj
```

Example datasets:  
Human lung microbiome   
```{r}
data(lungData)
lungData
```

Humanized gnotobiotic mouse gut  
```{r}
data(mouseData)
mouseData
```

Useful commands:  
Phenotype information can be accessed with the phenoData and pData methods:  
```{r}
phenoData(obj)
```

```{r}
head(pData(obj), 3)
```

Feature information can be accessed with the featureData and fData methods:
```{r}
featureData(obj)
```

```{r}
head(fData(obj)[, -c(2, 10)], 3)
```

The raw or normalized counts matrix can be accessed with the MRcounts function:
```{r}
head(MRcounts(obj[, 1:2]))
```

A MRexperiment-class object can be easily subsetted, for example:
```{r}
featuresToKeep = which(rowSums(obj) >= 100)
samplesToKeep = which(pData(obj)$SmokingStatus == "Smoker")
obj_smokers = obj[featuresToKeep, samplesToKeep]
obj_smokers
```

```{r}
head(pData(obj_smokers), 3)
```

Alternative normalization scaling factors can be accessed or replaced with the normFactors method:
```{r}
head(normFactors(obj))
```

```{r}
normFactors(obj) <- rnorm(ncol(obj))
head(normFactors(obj))
```

Library sizes (sequencing depths) can be accessed or replaced with the libSize method:
```{r}
head(libSize(obj))
```

```{r}
libSize(obj) <- rnorm(ncol(obj))
head(libSize(obj))
```

Additionally, data can be filltered to maintain a threshold of minimum depth or OTU presence:
```{r}
data(mouseData)
filterData(mouseData, present = 10, depth = 1000)
```

Two MRexperiment-class objects can be merged with the mergeMRexperiments function.
```{r}
data(mouseData)
newobj = mergeMRexperiments(mouseData, mouseData)
## MRexperiment 1 and 2 share sample ids; adding labels to sample ids.
newobj
```

Calculating normalization factors
```{r}
data(lungData)
p = cumNormStatFast(lungData)
# To calculate the scaling factors we simply run cumNorm
lungData = cumNorm(lungData, p = p)
```
There are other functions, including normFactors, cumNormMat, that return the normalization factors or a normalized matrix for a specified percentile

Calculating normalization factors using Wrench
```{r}
condition = mouseData$diet
mouseData = wrenchNorm(mouseData, condition = condition)
```

Exporting data  
To export normalized count matrices:
```{r}
mat = MRcounts(lungData, norm = TRUE, log = TRUE)[1:5, 1:5]
exportMat(mat, file = file.path(dataDirectory, "tmp.tsv"))
```

To save sample statistics (sample scaling factor, quantile value, number of identified features and library size):
```{r}
exportStats(lungData[, 1:5], file = file.path(dataDirectory, "tmp.tsv"))
## Default value being used.
head(read.csv(file = file.path(dataDirectory, "tmp.tsv"), sep = "\t"))
```

Example using fitFeatureModel for differential abundance testing
```{r}
data(lungData)
lungData = lungData[, -which(is.na(pData(lungData)$SmokingStatus))]
lungData = filterData(lungData, present = 30, depth = 1)
lungData <- cumNorm(lungData, p = 0.5)
pd <- pData(lungData)
mod <- model.matrix(~1 + SmokingStatus, data = pd)
lungres1 = fitFeatureModel(lungData, mod)
head(MRcoefs(lungres1))
```

Example using fitZig for differential abundance testing
```{r}
data(lungData)
controls = grep("Extraction.Control", pData(lungData)$SampleType)
lungTrim = lungData[, -controls]
rareFeatures = which(rowSums(MRcounts(lungTrim) > 0) < 10)
lungTrim = lungTrim[-rareFeatures, ]
lungp = cumNormStat(lungTrim, pFlag = TRUE, main = "Trimmed lung data")
## Default value being used.
```
  

```{r}
lungTrim = cumNorm(lungTrim, p = lungp)

## including body site as covariates and want to test for the bacteria 
## differentially abundant between smokers and nonsmokers

smokingStatus = pData(lungTrim)$SmokingStatus
bodySite = pData(lungTrim)$SampleType
normFactor = normFactors(lungTrim)
normFactor = log2(normFactor/median(normFactor) + 1)
mod = model.matrix(~smokingStatus + bodySite + normFactor)
settings = zigControl(maxit = 10, verbose = TRUE)
fit = fitZig(obj = lungTrim, mod = mod, useCSSoffset = FALSE,
control = settings)
```

Multiple groups  
```{r}
# maxit=1 is for demonstration purposes
settings = zigControl(maxit = 1, verbose = FALSE)
mod = model.matrix(~bodySite)
colnames(mod) = levels(bodySite)
# fitting the ZIG model
res = fitZig(obj = lungTrim, mod = mod, control = settings)
# The output of fitZig contains a list of various useful
# items. hint: names(res). Probably the most useful is the
# limma 'MLArrayLM' object called fit.
zigFit = slot(res, "fit")
finalMod = slot(res, "fit")$design
contrast.matrix = makeContrasts(BAL.A - BAL.B, OW - PSB, levels = finalMod)
fit2 = contrasts.fit(zigFit, contrast.matrix)
fit2 = eBayes(fit2)
topTable(fit2)
```

Exporting files
```{r}
taxa = sapply(strsplit(as.character(fData(lungTrim)$taxa), split = ";"), 
              function(i) {
  i[length(i)] 
                })
head(MRcoefs(fit, taxa = taxa, coef = 2))
```

Log Normal permutation test
```{r}
coeffOfInterest = 2
res = fitLogNormal(obj = lungTrim, mod = mod, useCSSoffset = FALSE,
B = 10, coef = coeffOfInterest)
# extract p.values and adjust for multiple testing res$p are
# the p-values calculated through permutation
adjustedPvalues = p.adjust(res$p, method = "fdr")
# extract the absolute fold-change estimates
foldChange = abs(res$fit$coef[, coeffOfInterest])
# determine features still significant and order by the
sigList = which(adjustedPvalues <= 0.05)
sigList = sigList[order(foldChange[sigList])]
# view the top taxa associated with the coefficient of
# interest.
head(taxa[sigList])
```

Presence-absence testing  
```{r}
classes = pData(mouseData)$diet
res = fitPA(mouseData[1:5, ], cl = classes)
# Warning - the p-value is calculating 1 despite a high odd's
# ratio.
head(res)
```

Discovery odds ratio testing  
```{r}
classes = pData(mouseData)$diet
res = fitDO(mouseData[1:100, ], cl = classes, norm = FALSE, log = FALSE)
head(res)
```

Feature correlations  
```{r}
cors = correlationTest(mouseData[55:60, ], norm = FALSE, log = FALSE)
head(cors)
```

Unique OTUs or features  
```{r}
cl = pData(mouseData)[["diet"]]
uniqueFeatures(mouseData, cl, nsamples = 10, nreads = 100)
```

Aggregating counts   
```{r}
obj = aggTax(mouseData, lvl = "phylum", out = "matrix")
head(obj[1:5, 1:5])
```

```{r}
obj = aggSamp(mouseData, fct = "mouseID", out = "matrix")
head(obj[1:5, 1:5])
```

Structural overview  
```{r}
trials = pData(mouseData)$diet
heatmapColColors = brewer.pal(12, "Set3")[as.integer(factor(trials))]
heatmapCols = colorRampPalette(brewer.pal(9, "RdBu"))(50)

# plotMRheatmap
plotMRheatmap(obj = mouseData, n = 200, cexRow = 0.4, cexCol = 0.4,
trace = "none", col = heatmapCols, ColSideColors = heatmapColColors)
# plotCorr
plotCorr(obj = mouseData, n = 200, cexRow = 0.25, cexCol = 0.25,
trace = "none", dendrogram = "none", col = heatmapCols)

cl = factor(pData(mouseData)$diet)
# plotOrd - can load vegan and set distfun = vegdist and use
# dist.method='bray'
plotOrd(mouseData, tran = TRUE, usePCA = FALSE, useDist = TRUE,
bg = cl, pch = 21)
# plotRare
res = plotRare(mouseData, cl = cl, pch = 21, bg = cl)
# Linear fits for plotRare / legend
tmp = lapply(levels(cl), function(lv) lm(res[, "ident"] ~ res[,
"libSize"] - 1, subset = cl == lv))
for (i in 1:length(levels(cl))) { 
  abline(tmp[[i]], col = i)
 }
legend("topleft", c("Diet 1", "Diet 2"), text.col = c(1, 2),
box.col = NA)
```

Feature specific
```{r}
head(MRtable(fit, coef = 2, taxa = 1:length(fData(lungTrim)$taxa)))
```

```{r}
patients = sapply(strsplit(rownames(pData(lungTrim)), split = "_"),
function(i) {
i[3]
})
pData(lungTrim)$patients = patients
classIndex = list(smoker = which(pData(lungTrim)$SmokingStatus ==
"Smoker"))
classIndex$nonsmoker = which(pData(lungTrim)$SmokingStatus ==
"NonSmoker")
otu = 779
# plotOTU
plotOTU(lungTrim, otu = otu, classIndex, main = "Neisseria meningitidis")
# Now multiple OTUs annotated similarly
x = fData(lungTrim)$taxa[otu]
otulist = grep(x, fData(lungTrim)$taxa)
# plotGenus
plotGenus(lungTrim, otulist, classIndex, labs = FALSE, main = "Neisseria meningitidis")
lablist <- c("S", "NS")
axis(1, at = seq(1, 6, by = 1), labels = rep(lablist, times = 3))

classIndex = list(Western = which(pData(mouseData)$diet == "Western"))
classIndex$BK = which(pData(mouseData)$diet == "BK")
otuIndex = 8770
# par(mfrow=c(1,2))
dates = pData(mouseData)$date
plotFeature(mouseData, norm = FALSE, log = FALSE, otuIndex, classIndex,
col = dates, sortby = dates, ylab = "Raw reads")
```

Summary  
metagenomeSeq is specifically designed for sparse high-throughput sequencing experiments
that addresses the analysis of differential abundance for marker-gene survey data. The package,
while designed for marker-gene survey datasets, may be appropriate for other sparse data sets
for which the zero-inflated Gaussian mixture model may apply.

Citing metagenomeSeq  
```{r}
citation("metagenomeSeq")
```






